tools/common/Driver.cs

```diff
@@ -67,6 +67,15 @@ static bool ParseOptions (Application app, Mono.Options.OptionSet options, strin
 					app.DebugAssemblies.Add (Path.GetFileName (v));
 				}
 			});
+			options.Add ("reference=", "Add an assembly to be processed.", v => app.References.Add (v));
+			// Unfortunately -r is used in mmp for something else (--resource), which means we can't use the same arguments for both mtouch and mmp.
+			// So add --reference, which is now used by both (and accepted by bgen as well), and deprecate -r|--ref for mtouch and -a|--assembly for mmp.
+#if MTOUCH
+			options.Add ("r|ref=", "Add an assembly to the resolver [DEPRECATED, use --reference instead]", v => app.References.Add (v), true);
+#endif
+#if MMP
+			options.Add ("a|assembly=", "Add an assembly to be processed [DEPRECATED, use --reference instead]", v => app.References.Add (v), true);
+#endif
```

Comment:
^ should not those remain the the tool-specific files to avoid `#if`?

Reply:
I guess it is fine to keep together and maybe be able to remove/hide in the future

Reply:
@spouliot Yes, the idea is to keep the related code close, but I'm fine either way.

Reply:
I've moved this code back into mtouch.cs/driver.cs